### ISSUE TITLE
[support.initlist] Reorganize like [support.srcloc]

### ISSUE THREAD

**Issue by JohelEGP** [cplusplus/draft#5720]
Originating comment: https://github.com/cplusplus/draft/pull/5713#discussion_r939552729.

Compare:
![1659968613](https://user-images.githubusercontent.com/21071787/183440751-2b4a9b1a-2e1a-4df3-8158-31b3f76ec55e.png)
![1659968626](https://user-images.githubusercontent.com/21071787/183440756-6eb33afd-788c-4293-8d68-bd8f0d6b5c1d.png)

What's titled as "Header `<initializer_list>` synopsis" is actually the class' synopsis. I suggest copying the format of [support.srcloc] instead.

**Comment by tkoeppe** [comment]
It looks like we've combined the header and the class synopsis here. Given how small the header is, what's the value in splitting the two? Is the status quo unclear or confusing?

**Comment by JohelEGP** [comment]
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1208r6.pdf used to look pretty much like [support.initlist] looks now, but that changed during https://github.com/cplusplus/draft/pull/3037.
![1660227190](https://user-images.githubusercontent.com/21071787/184153828-1834471f-add5-4c41-9427-a5e097ee33e7.png)

**Comment by tkoeppe** [comment]
Yes, I see. We had to fix the hanging paragraphs, but arguably source_location does that in a sligthly more principled way.

**Comment by tkoeppe** [comment]
On the flip side, source_location requires excessive nesting for such a short subclause, so I'm not sure which way is better here.
